tenpy/algorithms/dmrg.py

```diff
@@ -984,15 +951,19 @@ def update_LP(self, i0, U):
         i0 : int
             Site index. We calculate ``self.env.get_LP(i0+1)``.
         """
-        print("Setting LP at", i0 + 1)
-        if self.combine:
+        i0 = self.i0
+        if self.combine and self.move_right:
             LP = npc.tensordot(self.eff_H.LHeff, U, axes=['(vR.p*)', '(vL.p)'])
             LP = npc.tensordot(U.conj(), LP, axes=['(vL*.p*)', '(vR*.p)'])
-            self.env.set_LP(i0 + 1, LP, age=self.env.get_LP_age(i0) + 1)  # Always i0 + 1
+            self.env.set_LP(i0 + 1, LP, age=self.env.get_LP_age(i0) + 1)
         else:  # as implemented directly in the environment
-            self.env.get_LP(i0 + 1, store=True)
+            if self.move_right:
+                self.env.get_LP(i0 + 1, store=True)
+            else:
```

Comment:
If we are not updating LP on a left-move when `combine=True`, do we need to do it when `combine=False`?

Reply:
This function gets only called for local updates where `update_LP` defined in the `get_sweep_schedule`  is True, so it should *always* update the LP. 
In fact, if you look closely, it does get updated in either case, it's just that if we move left, the effective Hamiltonian doesn't have the `LHeff` part in single site, so it needs to call the self.env.get_LP directly. This does the same job, but needs to do one more tensordot contraction.